hazelcast-integration/spring-jcache/src/main/java/com/hazelcast/spring/jcache/SpringJCacheJavaConfig.java

```diff
@@ -0,0 +1,24 @@
+package com.hazelcast.spring.jcache;
+
+import com.hazelcast.client.HazelcastClient;
+import com.hazelcast.core.Hazelcast;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.annotation.AnnotationConfigApplicationContext;
+
+import static com.hazelcast.spring.jcache.DummyBeanTest.*;
+
+/**
+ * Example of Spring Caching with JCache
+ */
+public class SpringJCacheJavaConfig {
+    public static void main(String[] args) {
+        System.setProperty("hazelcast.jcache.provider.type", "client");
```

Comment:
I checked the code and now I think setting the property is necessary in your case. Because you do not create client explicitly. You get CachingProvider to create a client instance. Sample with XML configuration was a little different. It first created a server and a client and then it created cacheManager using already instantiated client. So I think it is OK.